src/chocolatey/infrastructure.app/services/NugetService.cs

```diff
@@ -1045,12 +1045,45 @@ private void set_package_names_if_all_is_specified(ChocolateyConfiguration confi
                 config.Input = string.Empty;
 
                 var localPackages = list_run(config, logResults: false);
+                var packagesToUpdate = localPackages.Select(p => p.Key).ToList();
+
+                if (!String.IsNullOrEmpty(config.UpgradeCommand.PackageNamesToSkip))
+                {
+                    var packagesToSkip = config.UpgradeCommand.PackageNamesToSkip
+                        .Split(',')
+                        .Select(x => x.Trim())
+                        .Where(x => !String.IsNullOrEmpty(x))
+                        .ToList();
+
+                    var unknownPackagesToSkip = packagesToSkip
+                        .Where(x => !packagesToUpdate.Contains(x, StringComparer.OrdinalIgnoreCase))
+                        .ToList();
+
+                    if (unknownPackagesToSkip.Any())
+                    {
+                        this.Log().Warn(() => "Some packages specified in '-except' were not found in the local packages: '{0}'".format_with(String.Join(",", unknownPackagesToSkip)));

```

Comment:
That makes sense. What about:
* `Some packages specified in the 'except' list were not found in the local packages: '{0}'` and;
* `These packages will not be upgraded because they were specified in the 'except' list: {0}`